jax/_src/interpreters/mlir.py

```diff
@@ -410,17 +404,49 @@ class ShapePolyLoweringState:
   # from an inner call to a polymorphic Exported.
   uses_dim_vars: bool
 
-  def __init__(self, dim_vars: tuple[str, ...],
-               lowering_platforms: tuple[str, ...]):
-    self.lowering_platforms = lowering_platforms
+  # If the first dimension variable is a platform index argument
+  has_platform_index_argument: bool
+
+  def __init__(self,
+               dim_vars: tuple[str, ...],
+               lowering_platforms: tuple[str, ...] | None):
     self.uses_dim_vars = (len(dim_vars) > 0)
-    if len(lowering_platforms) > 1:
+    if lowering_platforms is not None and len(lowering_platforms) > 1:
       dim_vars = ("platform_index_",) + tuple(dim_vars)
+      self.has_platform_index_argument = True
+    else:
+      self.has_platform_index_argument = False
     self.dim_vars = dim_vars
 
-  @property
-  def has_platform_index_argument(self):
-    return len(self.lowering_platforms) > 1
+
+dataclasses.dataclass(frozen=True)
+class LoweringParameters:
+  # A mapping between primitives and user-defined LoweringRules.
+  # When lowering a primitive, give priorioty to the rule in this map over
+  # existing Jax rules.
+  override_lowering_rules: tuple[tuple[core.Primitive, LoweringRule]] | None = None
+
+  # The lowering_platform, in case we do cross-platform lowering.
+  # One of 'cpu', 'cuda', 'rocm', 'tpu'.
+  # If None, use the default JAX mechanisms to pick the lowering platform.
+  # This is currently used for export and jax2tf.
+  lowering_platform: str | None
+
+  # The current lowering platforms, a non-empty tuple containing some of
+  # 'cpu', 'cuda', 'rocm', 'tpu'.
+  lowering_platforms: tuple[str, ...] | None
+
+  def __init__(self, *,
+               override_lowering_rules=None,
+               lowering_platform=None,
+               lowering_platforms=None):
+    self.override_lowering_rules = override_lowering_rules
+    self.lowering_platforms = lowering_platforms
+    assert lowering_platform is None or lowering_platforms is None
```

Comment:
If you allow `lowering_platforms` to be a `Sequence`, you could also remove `lower_platform` and require to always pass a sequence.

```
LoweringParameters(lowering_platforms=["gpu"])
```

is just as readable as

```
LoweringParameters(lowering_platform="gpu"])
```

I'd also recommend dropping the `lowering_` prefix, since the class name already includes `Lowering`.

Reply:
I dropped the lowering_ prefix in a few places. I changed the name of `lowering_platform` to `override_platform`